pooch/tests/test_processors.py

```diff
@@ -49,6 +49,35 @@ def test_decompress(method, ext):
         check_tiny_data(fname)
 
 
+def test_decompress_rename():
+    "Check that decompression rename customization works"
+    method = "auto"
+    ext = "gz"
+    processor = Decompress(method, lambda fname: fname[:-3])
+    with TemporaryDirectory() as local_store:
+        path = Path(local_store)
+        true_path = str(path / "tiny-data.txt")
```

Comment:
```suggestion
        fname = "tiny-data.txt"
        true_path = str(path / fname)
```

Reply:
Thank you, @santisoler . Indeed the possibility of name collision must be taken into account. Appending an extension as it is makes it very unlikely to happen.

One thing to consider as well is that decompressing files like "foo.txt.gz" by standard tools will result in a "foo.txt" file. So when I first used `pooch` I was surprised that it appended an extension instead of removing the extension regarding the compression method. From the philosophy of "not surprising the user", I think the current approach is not ideal. But I do understand your concerns about collision. Maybe a warning in the parameter documentation of `rename` might be enough.

I will apply your suggestions as soon as I can. Thanks again!